File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/exception/ExceptionTest.java
package ch.hsr.ifs.iltis.core.core.tests.exception;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import ch.hsr.ifs.iltis.core.core.exception.ILTISException;


public class ExceptionTest {

   @Rule
   public ExpectedException thrown = ExpectedException.none();

   @Test
   public void preservesExceptionMessage() {
      thrown.expect(ILTISException.class);
      thrown.expectMessage("Invalid XYZ");
      throw new ILTISException("Invalid XYZ").rethrowUnchecked();
   }

   @Test
   public void rethrowNestedExceptionWorks() throws Exception {
      thrown.expect(IllegalArgumentException.class);
      thrown.expectMessage("Number not in range");

      try {
         throw new ILTISException(new IllegalArgumentException("Number not in range"));
      }
      catch (final ILTISException e1) {
         e1.rethrow();
      }
   }
}

File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/exception/ExceptionTest.kt
package ch.hsr.ifs.iltis.core.core.tests.exception;

import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import ch.hsr.ifs.iltis.core.core.exception.ILTISException;


class ExceptionTest {

	@Rule
	@JvmField
	val thrown = ExpectedException.none();

	@Test
	fun preservesExceptionMessage() {
		thrown.expect(ILTISException::class.java);
		thrown.expectMessage("Invalid XYZ");
		throw ILTISException("Invalid XYZ").rethrowUnchecked();
	}

	@Test
	fun rethrowNestedExceptionWorks() {
		thrown.expect(IllegalArgumentException::class.java);
		thrown.expectMessage("Number not in range");

		try {
			throw ILTISException(IllegalArgumentException("Number not in range"));
		} catch (e1: ILTISException) {
			e1.rethrow();
		}
	}
}
